scene/gui/color_picker.cpp

```diff
@@ -1414,6 +1531,81 @@ void ColorPicker::_uv_input(const Ref<InputEvent> &p_event, Control *c) {
 			emit_signal(SNAME("color_changed"), color);
 		}
 	}
+
+	// TODO: Is it better to cast and check it or to not check it and get vector?
+	Ref<InputEventKey> kev = p_event;
+	Ref<InputEventJoypadButton> jbev = p_event;
+	Ref<InputEventJoypadMotion> jmev = p_event;
+
+	if (kev.is_valid() || jbev.is_valid() || jmev.is_valid()) {
+		// TODO: Consider adding new ui actions specific to ColorPicker, like the ones used for LineEdit.
+		Vector2 color_change_vector;
+		if (p_event->is_action_pressed("ui_left", true)) {
+			color_change_vector.x -= 1;
+		} else if (p_event->is_action_pressed("ui_right", true)) {
+			color_change_vector.x += 1;
+		} else if (p_event->is_action_pressed("ui_up", true)) {
+			color_change_vector.y -= 1;
+		} else if (p_event->is_action_pressed("ui_down", true)) {
+			color_change_vector.y += 1;
+		}
+
+		echo_multiplier = CLAMP(echo_multiplier * (p_event->is_echo() ? echo_multiplier_step : 1), 1, 25);
+
+		if (!color_change_vector.is_zero_approx()) {
+			color_change_vector *= echo_multiplier;
+
+			if (actual_shape == SHAPE_HSV_RECTANGLE) {
+				s = CLAMP(s + color_change_vector.x / 100.0, 0, 1);
+				v = CLAMP(v - color_change_vector.y / 100.0, 0, 1);
+			} else if (actual_shape == SHAPE_VHS_CIRCLE || actual_shape == SHAPE_OKHSL_CIRCLE) {
+				Vector2 center = c->get_size() / 2.0;
+
+				// HACK: It's a hack, as it messes up if I calculate it this way always.
```

Comment:
What does HACK refer to? `Vector2i()` is initial value of the cursor vector, so it needs to be initialized. Not sure what the other condition is for, it seems to have no effect.

Reply:
At first, I wanted to save the cursor position calculated in `_hsv_draw` and use it in `_uv_input`, but the cursor was moving in strange directions, as I mentioned in another comment. I checked if calculating it here every time would fix the issue, but it didn't. That's why I decided to calculate it only if it was not set (or was reset by clicking with mouse) and this is what the hack refers to, as it  is not the way I think it should be done, but it does not seem to work well with the correct way. I might 

> Not sure what the other condition is for, it can be safely removed it seems.
I added it when I was working on sliding along the edge of the circle when you keep pressing any direction. But it seems it is not needed now. Removed.